### ISSUE TITLE
Macro expansion is producing false positive E0308

### ISSUE THREAD

**Issue by mrjoe7** [intellij-rust/intellij-rust#9266]
## Environment

* **IntelliJ Rust plugin version:**
Rust plugin version: 0.4.177.4844-222
 
* **Rust toolchain version:**
rustc 1.64.0-nightly (4493a0f47 2022-08-02)

* **IDE name and version:**
CLion 2022.2.1
Build #CL-222.3739.54, built on August 16, 2022
Runtime version: 17.0.3+7-b469.37 amd64

* **Operating system:**
openSUSE Tumbleweed 20220823

## Problem description
IDE is showing false positive E0308 for sycamore `#[component]` macro 

## Steps to reproduce
Create new sycamore project and a component. IDE will be unable to correctly infer the return type:
![Screenshot_20220830_222642](https://user-images.githubusercontent.com/16738219/187536631-cba405e6-d408-4632-86c8-1abec4888703.png)

Minimal `main.rs` contents:
```rust
use sycamore::prelude::*;

#[component]
fn Hello<G: Html>(cx: Scope) -> View<G> {
    view! { cx,
        p { "Hello World!" }
    }
}
```

Minimal `cargo.toml` contents:
```toml
[package]
name = "sycamore-playground"
version = "0.1.0"
edition = "2021"

[dependencies]
sycamore = "0.8.0"
```

**Comment by Undin** [comment]
@mrjoe7 It seems it's because of `view!` proc macro call - the plugin doesn't expand procedural macro calls yet by default. But you still can try our procedural macro expander to fix your issue. See #6908 how to do it and for more details

**Comment by mrjoe7** [comment]
That actually worked pretty well. Now all the false positives are gone! Nice work, thank you!

**Comment by Undin** [comment]
Let's keep it open since procedural macro expansion is not enabled by default yet
